specs/protocol/tx_validity.md

```diff
@@ -84,14 +84,14 @@ For each asset ID `asset_id` in the input and output set:
 def sum_inputs(tx, col) -> int:
     total: int = 0
     for input in tx.inputs:
-        if input.type == InputType.Coin and input.asset_id == col:
+        if (input.type == InputType.Coin and input.asset_id == col) or (input.type == InputType.Message and col == 0):
             total += input.amount
     return total
 
 def sum_outputs(tx, col) -> int:
     total: int = 0
     for output in tx.outputs:
-        if (output.type == OutputType.Coin or output.type == OutputType.Withdrawal) and output.asset_id == col:
+        if (output.type == OutputType.Coin and output.asset_id == col) or (output.type == OutputType.Message and col == 0):
```

Comment:
wait, why would we include the amount of message outputs for `col == 0` if this method is calculating the total for `col == x`?

Reply:
This means if I sent some USDC in the same tx I withdraw, it would add the eth/fuel amount of the withdraw message to the USDC amount.

Reply:
Color 0 is the base asset right? An OutputMessage with an amount value is effectively a withdrawal/burn on the base asset and must be accounted for.

Reply:
The amount field is only used to indicate the base asset (ETH) being sent along with the message to the recipient. Like how you can send a value as part of an EVM call

Reply:
Ah ok, I think I misread as `output.asset_id == 0`. So this is just special casing output messages since they don't have variable asset_id's?

Reply:
Correct. The amount value is always for asset 0

Reply:
cool, disregard my comment then